src/cpp/core/include/core/http/Response.hpp

```diff
@@ -273,8 +274,24 @@ class Response : public Message
       if (request.acceptsEncoding(kGzipEncoding))
          setContentEncoding(kGzipEncoding);
       
-      // set body from file
-      Error error = setBody(filePath, filter);
+      Error error;
+
+      // Apply padding to fix QT short response issues
```

Comment:
Since we have access to the request here, can we use it to scope this behavior to Qt to reduce the risk of triggering problems elsewhere? There should be a user agent header on the request that identifies it as originating from QtWebKit.